packages/terra-application/src/page/Page.jsx

```diff
@@ -0,0 +1,136 @@
+   * component will be rendered below the Page's header.
+   */
+  toolbar: PropTypes.node,
+  /**
+   * A function to be executed upon selection of the Page's back button. If not
+   * provided, the back button will not be rendered.
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * When true, the Page will not prompt the user prior to executing
+   * `onRequestClose` in the presence of rendered UnsavedChangesPrompts.
+   * Use this prop to customize UnsavedChangesPrompt handling prior to Page
+   * closure.
+   */
+  dangerouslyDisableUnsavedChangesPromptHandling: PropTypes.bool,
+  /**
+   * The components to render within the Page.
+   */
+  children: PropTypes.node,
+};
+
+const Page = ({
+  label,
+  metaData,
+  actions,
+  toolbar,
+  onRequestClose,
+  dangerouslyDisableUnsavedChangesPromptHandling,
+  children,
+}) => {
+  const intl = React.useContext(ApplicationIntlContext);
+
+  if (actions && actions.type !== PageActions) {
+    throw new Error(`[terra-application] Page.Actions must be used to define actions for ${label}.`);
+  }
+
+  if (toolbar && toolbar.type !== PageToolbar) {
+    throw new Error(`[terra-application] Page.Toolbar must be used to define a toolbar for ${label}.`);
+  }
+
+  // An NavigationPromptCheckpoint is used to detect unsaved changes within
+  // the Page's content.
+  const checkpointRef = React.useRef();
+
+  // A Provider/Presenter pair is generated to manage NotificationBanner
+  // rendering within the Page.
+  const {
+    NotificationBannerProvider,
+    NotificationBanners,
+  } = useNotificationBanners();
+
+  // The usePagePortal hook is used to generate the PagePortal component that
+  // will render the Page content.
+  const { PagePortal, pageId } = usePagePortal({
+    label,
+    metaData,
+  });
+
+  return (
+    <PagePortal>
+      <div className={cx('page')}>
+        <div className={cx('header')}>
+          <PageHeader
+            id={pageId}
+            onSelectBack={onRequestClose ? () => {
+              if (dangerouslyDisableUnsavedChangesPromptHandling) {
+                onRequestClose();
+                return;
+              }
+
+              checkpointRef.current.resolvePrompts(getUnsavedChangesPromptOptions(intl)).then(() => {
+                onRequestClose();
+              });
+            } : undefined}
+            label={label}
+            actions={actions}
+            toolbar={toolbar}
+            NotificationBanners={NotificationBanners}
+          />
+        </div>
+        <div className={cx('content')}>
+          <NavigationPromptCheckpoint ref={checkpointRef}>
+            <NotificationBannerProvider>
+              {children}
+            </NotificationBannerProvider>
+          </NavigationPromptCheckpoint>
+        </div>
+      </div>
+    </PagePortal>
+  );
+};
+
+Page.propTypes = propTypes;
+Page.Actions = PageActions;
+Page.Action = PageAction;
+Page.Toolbar = PageToolbar;
```

Comment:
These might be a little controversial. As we have moved away from needing HOCs all over the place, the usage of static exports from some of our "umbrella" components becomes a little more straightforward and less error prone.

I feel like this simplifies the consumer's implementation code and keeps these components (that are required for usage with the Page) bundled together nicely. Take a look at the example implementations and let me know your thoughts.

Reply:
I like the explicit bundling of the page and it's sub components. My only concern is introducing inconsistency regarding how we present other API's in the package, such as workspace. It may be worth our time to go back and implement this style of api for those other areas at least in the terra-application package.

Reply:
Agreed, I can log an investigative story for that as part of our v2 release. We could remain passive by keeping any existing explicit exports in place.

Reply:
Logged issue: https://github.com/cerner/terra-application/issues/219